openvpn/middlewares/client/bytescount/stats_keeper.go

```diff
@@ -1,16 +1,53 @@
 package bytescount
 
+import (
+	"errors"
+	"time"
+)
+
 // SessionStatsKeeper keeps session stats
-type SessionStatsKeeper struct {
+type SessionStatsKeeper interface {
+	Save(stats SessionStats)
+	Retrieve() SessionStats
+	MarkSessionStart()
+	GetSessionDuration() (time.Duration, error)
+}
+
+// TimeGetter function returns current time
+type TimeGetter func() time.Time
+
+type sessionStatsKeeper struct {
 	sessionStats SessionStats
+	timeGetter   TimeGetter
+	sessionStart *time.Time
+}
+
+// NewSessionStatsKeeper returns new session stats keeper with given timeGetter function
+func NewSessionStatsKeeper(timeGetter TimeGetter) SessionStatsKeeper {
+	return &sessionStatsKeeper{timeGetter: timeGetter}
 }
 
 // Save saves session stats to keeper
-func (keeper *SessionStatsKeeper) Save(stats SessionStats) {
+func (keeper *sessionStatsKeeper) Save(stats SessionStats) {
 	keeper.sessionStats = stats
 }
 
 // Retrieve retrieves session stats from keeper
-func (keeper *SessionStatsKeeper) Retrieve() SessionStats {
+func (keeper *sessionStatsKeeper) Retrieve() SessionStats {
 	return keeper.sessionStats
 }
+
+// MarkSessionStart marks current time as session start time for statistics
+func (keeper *sessionStatsKeeper) MarkSessionStart() {
+	time := keeper.timeGetter()
+	keeper.sessionStart = &time
+}
+
+// GetSessionDuration returns elapsed time from marked session start
+func (keeper *sessionStatsKeeper) GetSessionDuration() (time.Duration, error) {
+	if keeper.sessionStart == nil {
```

Comment:
Do we really need to throw errors here and force any caller to handle them? It's just for statistics. Maybe we can simply return zero in case session is not started.

Reply:
Simplified to return zero in this case.